### PR TITLE
Automatic audio playback API

### PR DESCRIPTION
**Pull request by space-pope**
The Spokestack TTS client has been wrapped into a `TTSManager` subsystem along with a new output component that manages playback via a system media player. The playback component is lifecycle-aware; the only developer intervention it requires is being notified of changes in the current lifecycle owner (e.g., when a new Activity takes over).

(I also ran the IDE formatter/import optimizer on a few existing files during my explorations, resulting in a few random line break changes/deletions here and there.)

### CODE REVIEW

File: src/main/java/io/spokestack/spokestack/tts/TTSManager.java [hunk 1]
```diff
@@ -0,0 +1,304 @@
...
+            this.output = createComponent(this.outputClass, SpeechOutput.class);
+            this.ttsService.addListener(this.output);
+            this.output.setAppContext(this.context);
+            if (this.lifecycle != null) {
+                this.registerLifecycle(this.lifecycle);
+            }
+        }
+    }
+
+    private <T> T createComponent(String className, Class<T> clazz)
```
Comment (noelweichbrodt) [inline src/main/java/io/spokestack/spokestack/tts/TTSManager.java]: `clazz` 
shameless

File: src/main/java/io/spokestack/spokestack/tts/AudioResponse.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        this.metadata = responseData;
+        this.uri = audioUri;
+    }
+
+    /**
+     * Get additional data accompanying the synthesized audio.
+     *
+     * @return The response's metadata.
+     */
+    public Map<String, Object> getMetadata() {
```
[the remark this answers is not in the document]
Comment (space-pope) [reply src/main/java/io/spokestack/spokestack/tts/AudioResponse.java]: One of the core values in the Android library is pluggability/extensibility, so this and the `SynthesisRequest` object are tailored to Spokestack but written with the idea that the Spokestack API may require/return more data in the future, or a developer may choose to use a different TTS service, so these generic maps let them pass/receive any extra data they need without having to resort to extreme measures to write a new TTS service or Output component.

File: src/main/java/io/spokestack/spokestack/tts/SpokestackTTSOutput.java [hunk 3]
```diff
@@ -0,0 +1,348 @@
...
+     *               for participation in the TTS subsystem.
+     */
+    public SpokestackTTSOutput(@Nullable SpeechConfig config) {
+        // the constant value here is the same in ExoPlayer as in
+        // AudioAttributesCompat in versions 2.11.0 and v1.1.0, respectively
+        this.contentType = C.CONTENT_TYPE_SPEECH;
+
+        this.playerState = new PlayerState();
+
+        // 26 == Android O; we just don't want to pull in the extra
```
[the remark this answers is not in the document]
Comment (space-pope) [reply src/main/java/io/spokestack/spokestack/tts/SpokestackTTSOutput.java]: I had a similar thought.

File: src/main/java/io/spokestack/spokestack/tts/SpokestackTTSOutput.java [hunk 4]
```diff
@@ -0,0 +1,348 @@
...
+            player.addListener(this);
+            this.mediaPlayer = player;
+        });
+    }
+
+    @Override
+    public void close() {
+        this.taskHandler.accept(() -> {
+            savePlayerSettings();
+            this.mediaPlayer.release();
```
[the remark this answers is not in the document]
Comment (space-pope) [reply src/main/java/io/spokestack/spokestack/tts/SpokestackTTSOutput.java]: Me too. Something something singleton hardware codecs used by native code something something.

File: src/main/java/io/spokestack/spokestack/tts/SpokestackTTSOutput.java [hunk 5]
```diff
@@ -0,0 +1,348 @@
...
+
+    @Override
+    public void onStop(@NonNull LifecycleOwner owner) {
+        pauseContent();
+    }
+
+    /**
+     * Start or resume playback of any TTS responses.
+     */
+    public void playContent() {
```
[the remark this answers is not in the document]
Comment (space-pope) [reply src/main/java/io/spokestack/spokestack/tts/SpokestackTTSOutput.java]: I do too, but the system can snatch audio focus away from an integrating app for any number of reasons—an incoming phone call, a notification ding, voice accessibility features reading an incoming text, etc. In such cases, we want Spokestack audio to pause, not duck or continue playing, and resume when focus is re-granted. You're also "supposed" to pause/destroy media players/resume on lifecycle events like rotation and activity transitions.

Handling playback automatically commits the library to being a good citizen in the overall ecosystem of device audio.

File: src/main/java/io/spokestack/spokestack/tts/TTSManager.java (hunk 1 above)
Comment (space-pope) [reply src/main/java/io/spokestack/spokestack/tts/TTSManager.java]: The convention is shameless; I merely follow it.

**Pull request closed without merging**